lib/createlocaltracks.ts

```diff
@@ -149,6 +150,12 @@ export async function createLocalTracks(options?: CreateLocalTracksOptions): Pro
     extraLocalTrackOptions.audio.workaroundWebKitBug1208516 = fullOptions.audio.workaroundWebKitBug1208516;
   }
 
+  let noiseCancellationOptions: NoiseCancellationOptions|undefined;
+  if (typeof fullOptions.audio === 'object' && 'noiseCancellationOptions' in fullOptions.audio) {
+    noiseCancellationOptions = fullOptions.audio.noiseCancellationOptions;
+    delete fullOptions.audio.noiseCancellationOptions;
+  }
+
```

Comment:
We just kind of adapted this pattern in this file when we were converting it to TS. But now that we're building on top of it, do you think there's a better way to pass around options?

Reply:
yes, I followed the same pattern, but yeah, I will look into cleaning up this and existing code around this.

Reply:
This PR is already getting pretty big, and this change is following existing pattern. I need more thinking on changing the patter, but I will leave that for future PRs,